source/app/util/custom_resource.py

```diff
@@ -169,19 +172,20 @@ def handle_request(self):
         except Exception as ex:
             self.response["Status"] = "FAILED"
             self.response["Reason"] = str(ex)
-
-        timer.cancel()
-        return self._send_response()
+        finally:
+            timer.cancel()
+            return self._send_response()
```

Comment:
I'm finding it hard to reason about the flow of this function. What I have done for custom resource providers is something like:

```py
try:
    # do work, raise Exception on error
    # send response with successful status
except:
    # send response with failed status
```